additions to network topolgy

IC-encipher opened the pull request:
added graphs and explanations of graphs to better explain what a neighbourhood looks like and examples of what clients would be connected to if a neighbourhood is not properly formed.

IC-encipher committed:
grammer edit

IC-encipher committed:
update network topology

have add notes explaining that you can only communicate with people who have a direct connect your server, either by being on your server or their server is directly connected to yours

IC-encipher committed:
Update network topology

moved a sentence to a better location

IC-encipher committed:
Update readme.md

missing word in sentance

IC-encipher committed:
updated topology section

added diagrams showing examples of neighbourhoods, and added the explanation to the original 3 diagrams that they are an example of what happens when a neighbourhood isn't properly formed and who could talk to who in that sitatuion

IC-encipher committed:
Update network topology 

word change

IC-encipher committed:
Update network topology 

updating graphs

IC-encipher committed:
update network topology 

fixed grammar and spelling errors

IC-encipher committed:
Merge pull request #1 from Theo-encipher/network-topology-edit

Network topology section edit

IC-encipher committed:
Merge branch 'xvk-64:main' into main

IC-encipher committed:
Merge branch 'xvk-64:main' into network-topology-edit

xvk-64 commented:
This is a lot of text! I agree that additional examples would be useful for understanding, but maybe put them in an additional file?

IC-encipher commented:
will do

IC-encipher commented:
would it be worth leaving one example of a neighbourhood in the main spec document for example purposes also or should I move all the graphics?

IC-encipher committed:
Create network topology examples files

IC-encipher committed:
update network topology section

remove examples to add separate file, to help keep spec short and easier to read/follow

IC-encipher committed:
Update network topology examples

added examples copied over from spec

IC-encipher committed:
Rename network topology examples to network topology examples.md

IC-encipher commented:
have moved graphics to new file

IC-encipher committed:
Merge pull request #2 from IC-encipher/network-topology-edit

Network topology edit merge with main

IC-encipher commented:
should be visible in the commits now

xvk-64 commented on readme.md:
```diff
@@ -30,9 +30,10 @@ Server (Owner of the receiver)
 
-The transport layer of this protocol uses Websockets (RFC 6455)
+You can call the server you are connected to your homeserver. You can only connect to users connected to your home server or users on servers directly connected your home server. Refer to the additional file network topology examples, to see examples of neighbourhoods and other potential network arrangments. 
+To illustrate this consider these the diagrams below:
```
Forgot to remove this

IC-encipher committed:
remove unneeded sentance

IC-encipher commented:
removed 
To illustrate this consider these the diagrams below:

xvk-64 commented:
OK
